Signal/src/UIView+OWS.h

```diff
@@ -60,6 +60,31 @@ CGFloat ScaleFromIPhone5(CGFloat iPhone5Value);
 
 - (void)centerOnSuperview;
 
+#pragma mark - RTL
+
+// For correct right-to-left layout behavior, use "leading" and "trailing",
+// not "left" and "right".
+//
+// NOTE: the margin values are inverted in RTL layouts.
+- (BOOL)isRTL;
+- (NSArray<NSLayoutConstraint *> *)autoPinLeadingAndTrailingToSuperview;
+- (NSLayoutConstraint *)autoPinLeadingToSuperView;
+- (NSLayoutConstraint *)autoPinLeadingToSuperViewWithMargin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinTrailingToSuperView;
+- (NSLayoutConstraint *)autoPinTrailingToSuperViewWithMargin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinLeadingToTrailingOfView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinLeadingToTrailingOfView:(UIView *)view margin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinLeadingToView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinLeadingToView:(UIView *)view margin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinTrailingToView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinTrailingToView:(UIView *)view margin:(CGFloat)margin;
+// Return Right on LTR and Right on RTL.
+- (NSTextAlignment)textAlignmentUnnatural;
+// Leading and trailing anchors honor layout margins.
+// When using a UIView as a "div" to structure layout, we don't want it to have margins.
++ (UIView *)containerView;
+- (void)setHLayoutMargins:(CGFloat)value;
```

Comment:
> I'm not convinced yet that we need these... Can you point me to a piece of code where we use these new helpers where it can't be reasonably expressed with the existing autoLayout api?

Our custom methods are more concise - easier to read and easier to auto-complete than existing methods - and I _love_ small improvements in readability of layout code which is so hard to read.  

But the real difference is that our methods use `layoutMarginsGuide` anchors, which behave differently than the PureLayout alternatives you indicated.  Honoring `layoutMargins` is particularly important in cell layouts, where it lets us align with the complicated built-in behavior of table and collection view cells' default contents.